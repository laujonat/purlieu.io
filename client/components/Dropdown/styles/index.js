import styled from "styled-components"
import { fonts, colors } from "../../../lib/styles"

export const Wrapper = styled.div`
  font-size: ${fonts.small};
  margin-top: 1em;
  display: flex;
  justify-content: space-between;
  width: 50%;
`

export const Container = styled.div`
  width: 100%;
`

export const Header = styled.div`
  display: flex;
  justify-content: space-between;
  background-color: white;
  border-radius: 3px 3px 2px 2px;
  border: 1px solid ${colors.grey};
  cursor: pointer;
`

export const List = styled.ul`
  margin: 0 auto;
  text-align: center;
  background-color: white;
  list-style-type: none;
  border-radius: 2px;
`

export const CarrierOptions = styled.li`
  height: 20px;
  text-align: center;
  &:hover {
    background-color: ${colors.purplishGrey};
    border: 1px solid ${colors.grey};
    opacity: 0.8;
  }

  cursor: pointer;
`

export const CarrierLabel = styled.label`
  display: flex;
  align-self: flex-start;
  padding: 0.5em 1em;
  cursor: pointer;
`

export const IconContainer = styled.div`
  margin: auto 0.5em;
`
