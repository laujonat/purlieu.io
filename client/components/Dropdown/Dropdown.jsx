import React, { Component } from "react"
import styled from "styled-components"
import PropTypes from "prop-types"
import { fonts, colors } from "../../lib/styles"

const List = styled.select`
  margin: 0 auto;
  text-align: center;
  height: 50px;
  background-color: white;
  box-shadow: 0 0 20px rgba(0, 0, 0, 0.3);
`
const Label = styled.label`
  font-size: ${fonts.base};
`

const CarrierLabel = styled(Label)`
  display: flex;
  flex-direction: column;
  align-self: flex-start;
`

const Option = styled.option`
  text-align: center;
  height: 20px;
`

class Dropdown extends Component {
  render() {
    const { list, onChange, selectedCarrier, carrierLabel } = this.props
    return (
      <CarrierLabel>
        {carrierLabel}
        <List value={selectedCarrier} onChange={onChange}>
          {list.map((el, idx) => (
            <Option key={idx} value={el}>
              {el}
            </Option>
          ))}
        </List>
      </CarrierLabel>
    )
  }
}

Dropdown.propTypes = {
  list: PropTypes.arrayOf(PropTypes.string),
  selectedCarrier: PropTypes.string,
  onChange: PropTypes.func,
  carrierLabel: PropTypes.string
}

Dropdown.defaultProps = {
  list: [],
  onChange: () => {}
}

export default Dropdown
