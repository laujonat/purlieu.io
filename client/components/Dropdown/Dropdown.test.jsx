import React from "react"
import { shallow } from "enzyme"
import Dropdown from "./Dropdown"

describe("Dropdown", () => {
  it("renders the component", () => {
    const wrapper = shallow(<Dropdown />)
    expect(wrapper).toMatchSnapshot()
  })
})
