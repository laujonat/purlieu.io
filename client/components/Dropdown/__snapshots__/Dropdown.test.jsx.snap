// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders the component 1`] = `ShallowWrapper {}`;
